src/cirrus/management/deployment_pointer.py

```diff
@@ -97,68 +36,115 @@ class Pointer:
     def from_string(cls: type[Self], string: str) -> Self:
         obj = json.loads(string)
         obj["_type"] = obj.pop("type")
+
         return cls(**obj)
 
-    def resolve(self) -> PointerObject:
+    def resolve(self):
         match self._type:
-            case PointerType.SECRET:
-                return SecretArn.from_string(self.value)
+            case "parameter_store":
+                return ParamStoreDeployment(self.value)
             case _ as unknown:
                 raise TypeError(f"Unsupported pointer type '{unknown}'")
 
 
+class ParamStoreDeployment:
+    def __init__(self, deployment_key: str):
+        self.prefix = deployment_key
+
+    def fetch(self, session=boto3.Session) -> dict[str, str]:
+        """Get all parameters for specific deployment"""
+
+        parameters = DeploymentPointer._get_parameters(self.prefix, session)
+        return {
+            param["Name"].split(self.prefix)[1]: param["Value"] for param in parameters
```

Comment:
If you use my generator rewrite of `get_parameters_by_path()` below, then you can make this like:

```suggestion
            param["Name"].removeprefix(self.prefix): param["Value"]
            for param in get_parameters_by_path(self.prefix, session)
```

Doing so avoids the intermediate allocation of the entire list. Also note the use of the string method `removeprefix` as opposed to `split`.

Reply:
Yeah I'm really liking `yield` helping avoid those intermediate assignments.